File: src/main/java/com/nicholasnassar/sfsapi/data/assignments/AssignmentsWeek.java
package com.nicholasnassar.sfsapi.data.assignments;

import io.vertx.core.json.JsonArray;

import java.util.List;

public class AssignmentsWeek {
    private final List<AssignmentDay> days;

    public AssignmentsWeek(List<AssignmentDay> days) {
        this.days = days;
    }

    public JsonArray asJsonArray() {
        JsonArray array = new JsonArray();

        for (AssignmentDay day : days) {
            array.add(day.asJson());
        }

        return array;
    }
}

File: src/main/java/com/nicholasnassar/sfsapi/data/assignments/AssignmentsWeek.kt
package com.nicholasnassar.sfsapi.data.assignments

import io.vertx.core.json.JsonArray

class AssignmentsWeek(val days: List<AssignmentDay>) {
    fun asJsonArray(): JsonArray {
        val array = JsonArray()

        for (day in days) {
            array.add(day.asJson())
        }

        return array
    }
}
